### PR TITLE
fix: fixes a regex problem on groundswell manifest

### PR DESCRIPTION
**Pull request by justinegeffen**
Recreating PR https://github.com/seqeralabs/docs_old/pull/115

### CODE REVIEW

**Comment by justinegeffen** [comment]
@gwright99, please review this when you have a chance. Thank you!

**Comment by gwright99** [comment]
I'm a bit foggy on this issue (_looking at original ticket now - I remember the issue, I just don't remember exactly how to test_) -- any suggestions on how your folks tested it?

**Comment by justinegeffen** [comment]
@bebosudo, would you be open to reviewing this? Thank you!

**Comment by justinegeffen** [comment]
> I'm a bit foggy on this issue (_looking at original ticket now - I remember the issue, I just don't remember exactly how to test_) -- any suggestions on how your folks tested it?

I've added @bebosudo to this issue for that feedback as I believe he tested it. :)

File: platform_versioned_docs/version-23.3.0/enterprise/_templates/k8s/groundswell.yml [hunk 1]
```diff
@@ -68,7 +68,7 @@ spec:
             - |
               echo "$(date): starting check for db $TOWER_DB_URL"
               # Strip initial Java connection string and options after '?' from URI
-              until mysqlsh --uri "$(echo $TOWER_DB_URL |sed -e 's@jdbc:\(.*\)?.*@\1@')" -u"$TOWER_DB_USER" -p"$TOWER_DB_PASSWORD" --sql -e "SELECT VERSION()"; do
+              until mysqlsh --uri "$(echo $TOWER_DB_URL |sed -e 's@jdbc:\(.*\)@\1@')" -u"$TOWER_DB_USER" -p"$TOWER_DB_PASSWORD" --sql -e "SELECT VERSION()"; do
```
Comment (bebosudo) [inline platform_versioned_docs/version-23.3.0/enterprise/_templates/k8s/groundswell.yml:71]:
```suggestion
              until mysqlsh --uri "$(echo $TOWER_DB_URL |cut -d'?' -f1 |sed -e 's@jdbc:\(.*\)@\1@')" -u"$TOWER_DB_USER" -p"$TOWER_DB_PASSWORD" --sql -e "SELECT VERSION()"; do
```

File: platform_versioned_docs/version-23.4.0/enterprise/_templates/k8s/groundswell.yml [hunk 2]
```diff
@@ -68,7 +68,7 @@ spec:
             - |
               echo "$(date): starting check for db $TOWER_DB_URL"
               # Strip initial Java connection string and options after '?' from URI
-              until mysqlsh --uri "$(echo $TOWER_DB_URL |sed -e 's@jdbc:\(.*\)?.*@\1@')" -u"$TOWER_DB_USER" -p"$TOWER_DB_PASSWORD" --sql -e "SELECT VERSION()"; do
+              until mysqlsh --uri "$(echo $TOWER_DB_URL |sed -e 's@jdbc:\(.*\)@\1@')" -u"$TOWER_DB_USER" -p"$TOWER_DB_PASSWORD" --sql -e "SELECT VERSION()"; do
```
Comment (bebosudo) [inline platform_versioned_docs/version-23.4.0/enterprise/_templates/k8s/groundswell.yml:71]:
```suggestion
              until mysqlsh --uri "$(echo $TOWER_DB_URL |cut -d'?' -f1 |sed -e 's@jdbc:\(.*\)@\1@')" -u"$TOWER_DB_USER" -p"$TOWER_DB_PASSWORD" --sql -e "SELECT VERSION()"; do
```

**Review by bebosudo — CHANGES_REQUESTED** [review]
Unfortunately the proposed fix doesn't work when the tower db url contains extra parameters (like 99% of the cases) and only works with a db url without parameters like for the customer who pointed it out (seems to be JnJ from [this](https://github.com/seqeralabs/docs_old/pull/115#issuecomment-2013234355)?), as I showed in [this message](https://github.com/seqeralabs/docs_old/pull/115#issuecomment-2013537439):

```bash
[alberto@fedora]$ podman run --rm -it  --entrypoint bash mysql:5.7
bash-4.2#
bash-4.2# TOWER_DB_URL="jdbc:mysql://tower.cluster-c1rjv6umdpvw.eu-west-2.rds.amazonaws.com:3306/tower?&usePipelineAuth=false&useBatchMultiSend=false"
bash-4.2# mysqlsh --uri "$(echo $TOWER_DB_URL |sed -e 's@jdbc:\(.*\)@\1@')"
Invalid URI: Invalid connection option ''.

bash-4.2# TOWER_DB_URL="jdbc:mysql://tower.cluster-c1rjv6umdpvw.eu-west-2.rds.amazonaws.com:3306/tower"
bash-4.2# mysqlsh --uri "$(echo $TOWER_DB_URL |sed -e 's@jdbc:\(.*\)@\1@')"
Please provide the password for 'root@tower.cluster-c1rjv6umdpvw.eu-west-2.rds.amazonaws.com:3306': 
```

That's because mysqlsh doesn't accept extra parameters.

I proposed a fix that should work for both cases, your feedback would be appreciated @swampie @gwright99

```bash
bash-4.2# TOWER_DB_URL="jdbc:mysql://tower.cluster-c1rjv6umdpvw.eu-west-2.rds.amazonaws.com:3306/tower?&usePipelineAuth=false&useBatchMultiSend=false" 
bash-4.2# mysqlsh --uri "$(echo $TOWER_DB_URL |cut -d'?' -f1 |sed -e 's@jdbc:\(.*\)@\1@')"
Please provide the password for 'root@tower.cluster-c1rjv6umdpvw.eu-west-2.rds.amazonaws.com:3306':

bash-4.2# TOWER_DB_URL="jdbc:mysql://tower.cluster-c1rjv6umdpvw.eu-west-2.rds.amazonaws.com:3306/tower"
bash-4.2# mysqlsh --uri "$(echo $TOWER_DB_URL |cut -d'?' -f1 |sed -e 's@jdbc:\(.*\)@\1@')"
Please provide the password for 'root@tower.cluster-c1rjv6umdpvw.eu-west-2.rds.amazonaws.com:3306': 
```

**Comment by gwright99** [comment]
Given the recent addition of a mandatory connection string modifier for v24.1+ (and all the various connection string modifiers that need to be used for various DB products < 24.1), I think we need to scrap the old way and go with @bebosudo's new approach.

**Comment by justinegeffen** [comment]
Thanks, @bebosudo! Much appreciated. @gwright99, please review/approve when you have a chance. Thank you!

**Comment by justinegeffen** [comment]
@bebosudo, following up on this for approval. Thank you! :)

File: platform_versioned_docs/version-23.3/enterprise_templates/k8s/groundswell.yml [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+        app: groundswell
+    spec:
+      imagePullSecrets:
+        - name: "cr.seqera.io"
+      securityContext:
+        fsGroup: 101
+
+      initContainers:
+        - name: wait-for-tower-db
+          image: "mysql:5.7"
```
Comment (bebosudo) [inline platform_versioned_docs/version-23.3/enterprise_templates/k8s/groundswell.yml:64]:
```suggestion
          image: "mysql:8.0"
```

File: platform_versioned_docs/version-23.3/enterprise_templates/k8s/groundswell.yml [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+              echo "$(date): db server ready"
+          env:
+            - name: SLEEP_PERIOD_SECONDS
+              value: "5"
+          envFrom:
+            - configMapRef:
+                name: tower-backend-cfg
+
+        - name: wait-for-swell-db
+          image: "mysql:5.7"
```
Comment (bebosudo) [inline platform_versioned_docs/version-23.3/enterprise_templates/k8s/groundswell.yml:84]:
```suggestion
          image: "mysql:8.0"
```

File: platform_versioned_docs/version-23.3/enterprise_templates/k8s/groundswell.yml [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+  strategy:
+    type: Recreate
+
+  template:
+    metadata:
+      labels:
+        app: groundswell
+    spec:
+      imagePullSecrets:
+        - name: "cr.seqera.io"
```
Comment (bebosudo) [inline platform_versioned_docs/version-23.3/enterprise_templates/k8s/groundswell.yml:58]: I didn't check, but are there instructions in the previous steps on how to create this image pull secret?

File: platform_versioned_docs/version-23.3/enterprise_templates/k8s/groundswell.yml [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+              echo "$(date): db server ready"
+          env:
+            - name: SLEEP_PERIOD_SECONDS
+              value: "5"
+          envFrom:
+            - configMapRef:
+                name: tower-groundswell-cfg
+
+        - name: migrate-db
+          image: "cr.seqera.io/private/nf-tower-enterprise/groundswell:0.3.3"
```
Comment (bebosudo) [inline platform_versioned_docs/version-23.3/enterprise_templates/k8s/groundswell.yml:104]:
```suggestion
          image: "cr.seqera.io/private/nf-tower-enterprise/groundswell:0.4.0"
```

File: platform_versioned_docs/version-23.3/enterprise_templates/k8s/groundswell.yml [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+          command: ['/opt/groundswell/bin/migrate-db.sh']
+          envFrom:
+            - configMapRef:
+                name: tower-groundswell-cfg
+            - configMapRef:
+                name: tower-backend-cfg
+
+      containers:
+        - name: groundswell
+          image: "cr.seqera.io/private/nf-tower-enterprise/groundswell:0.3.3"
```
Comment (bebosudo) [inline platform_versioned_docs/version-23.3/enterprise_templates/k8s/groundswell.yml:114]:
```suggestion
          image: "cr.seqera.io/private/nf-tower-enterprise/groundswell:0.4.0"
```

File: platform_versioned_docs/version-23.3/enterprise_templates/k8s/groundswell.yml (hunk 6 above)
Comment (bebosudo) [reply platform_versioned_docs/version-23.3/enterprise_templates/k8s/groundswell.yml:104]: Don't know if this has been upgraded in another PR?

File: platform_versioned_docs/version-23.4/enterprise_templates/k8s/groundswell.yml [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+              echo "$(date): db server ready"
+          env:
+            - name: SLEEP_PERIOD_SECONDS
+              value: "5"
+          envFrom:
+            - configMapRef:
+                name: tower-groundswell-cfg
+
+        - name: migrate-db
+          image: "cr.seqera.io/private/nf-tower-enterprise/groundswell:0.3.3"
```
Comment (bebosudo) [inline platform_versioned_docs/version-23.4/enterprise_templates/k8s/groundswell.yml:104]: Similarly here and down below

**Review by bebosudo — COMMENTED** [review]
the regex looks correct now, I didn't test it

non-blocking: left some comments about possible improvements we could do, don't know if you prefer to follow up in a separate PR, up to you

File: platform_versioned_docs/version-23.3/enterprise_templates/k8s/groundswell.yml (hunk 5 above)
Comment (justinegeffen) [reply platform_versioned_docs/version-23.3/enterprise_templates/k8s/groundswell.yml:58]: The instructions are provided here: https://docs.seqera.io/platform/24.1/enterprise/kubernetes#pipeline-optimization.

**Comment by justinegeffen** [comment]
Thanks, @bebosudo! I updated this for 24.1 as well.

**Pull request merged**